src/lib.rs

```diff
@@ -523,22 +523,22 @@ fn json_arr_append(ctx: &Context, args: Vec<String>) -> RedisResult {
             .map_err(|e| e.into())
         })
         .map(|v: RedisValue| {
-            ctx.notify_keyspace_event(NotifyEvent::MODULE, "json.arrappend", key.as_str());
+            ctx.notify_keyspace_event(NotifyEvent::MODULE, "json.arrappend", &key);
             ctx.replicate_verbatim();
             v
         })
 }
 
 fn do_json_arr_append<I>(args: I, value: &mut Value) -> Result<Value, Error>
 where
-    I: Iterator<Item = String>,
+    I: Iterator<Item = RedisString>,
 {
     if !value.is_array() {
         return Err(err_json(value, "array"));
     }
 
     let mut items: Vec<Value> = args
-        .map(|json| serde_json::from_str(&json))
+        .map(|json| serde_json::from_str(&json.into_string_lossy()))
```

Comment:
Is a lossy conversion what we want here? I know the Redis way is not to return errors, but can't this cause silent data corruption in case the input is not valid UTF-8? JSON should always be valid UTF-8.